src/mlpack/methods/ann/layer/spatial_dropout.hpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file methods/ann/layer/spatial_dropout.hpp
+ * @author Anjishnu Mukherjee
+ *
+ * Definition of the SpatialDropout class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SPATIAL_DROPOUT_HPP
+#define MLPACK_METHODS_ANN_LAYER_SPATIAL_DROPOUT_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/dists/bernoulli_distribution.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the SpatialDropout layer.
+ *
+ * For more information, refer to the following paper,
+ *
+ * @code
+ * @article{Tompson15,
+ *   author    = {Jonathan Tompson, Ross Goroshin, Arjun Jain,
+ *                Yann LeCun, Christopher Bregler},
+ *   title     = {Efficient Object Localization Using Convolutional Networks},
+ *   journal   = {CoRR},
+ *   volume    = {abs/1411.4280},
+ *   year      = {2015},
+ *   url       = {https://arxiv.org/abs/1411.4280},
+ *   eprint    = {1411.4280},
+ * }
+ * @endcode
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class SpatialDropout
+{
+ public:
+  //! Create the SpatialDropout object.
+  SpatialDropout();
+  /**
+   * Create the SpatialDropout object using the specified parameters.
+   *
+   * @param size The number of channels of each input image.
+   * @param ratio The probability of each channel getting dropped.
+   */
+  SpatialDropout(size_t size, double ratio = 0.5);
```

Comment:
Just to be consistent with the rest of the codebase I would use `const` for each parameter here.